AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/DataStoreCustomPrimaryKeyTests.swift

```diff
@@ -0,0 +1,109 @@
+
+    func testCreateQueryDelete() throws {
+
+        try startAmplifyAndWaitForSync()
+        let customerOrder = CustomerOrder(orderId: UUID().uuidString, email: "test@abc.com")
+
+        let createReceived = expectation(description: "Create notification received")
+        let deleteReceived = expectation(description: "Delete notification received")
+
+        let hubListener = Amplify.Hub.listen(
+            to: .dataStore,
+            eventName: HubPayload.EventName.DataStore.syncReceived) { payload in
+                guard let mutationEvent = payload.data as? MutationEvent
+                    else {
+                        XCTFail("Can't cast payload as mutation event")
+                        return
+                }
+
+                guard let order = try? mutationEvent.decodeModel() as? CustomerOrder,
+                      order.id == customerOrder.id else {
+                    return
+                }
+
+                if mutationEvent.mutationType == GraphQLMutationType.create.rawValue {
+                    XCTAssertEqual(order.email, customerOrder.email)
+                    XCTAssertEqual(order.orderId, customerOrder.orderId)
+                    XCTAssertEqual(mutationEvent.version, 1)
+                    createReceived.fulfill()
+                    return
+                }
+
+                if mutationEvent.mutationType == GraphQLMutationType.delete.rawValue {
+                    XCTAssertEqual(order.email, customerOrder.email)
+                    XCTAssertEqual(order.orderId, customerOrder.orderId)
+                    XCTAssertEqual(mutationEvent.version, 2)
+                    deleteReceived.fulfill()
+                    return
+                }
+        }
+
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
+            XCTFail("Listener not registered for hub")
+            return
+        }
+
+        // create customer order
+        Amplify.DataStore.save(customerOrder) { _ in }
+        wait(for: [createReceived], timeout: networkTimeout)
+
+        // query the created order
+        let queryBeforeDeleteExpectation = expectation(description: "Queried model should be same as created one")
+        Amplify.DataStore.query(CustomerOrder.self, byId: customerOrder.id) { result in
+            switch result {
+            case .success(let value):
+                guard let order = value else {
+                    XCTFail("Queried model is nil")
+                    return
+                }
+                XCTAssertEqual(order.id, customerOrder.id)
+                XCTAssertEqual(order.orderId, customerOrder.orderId)
+                XCTAssertEqual(order.email, customerOrder.email)
+                queryBeforeDeleteExpectation.fulfill()
+            case .failure(let error):
+                print("Error : \(error)")
+            }
+        }
+        wait(for: [queryBeforeDeleteExpectation], timeout: networkTimeout)
```

Comment:
can we incorporate another DataStore.save to get an update mutation going? Ensuring update mutation works is important even though currently the update mutation has the same API signature of the create mutation

The other reason why we shoud start testing update mutations as well is looking forward to the work for "per changed fields model updates". When we do that work, it will be closely tied to custom primary keys since a model with only changed fields will include both the `id` and `orderId` regardless of which fields were changed.